policybot/handlers/githubwebhook/filters/unstale/unstale.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unstale
+
+import (
+	"context"
+
+	"github.com/google/go-github/v26/github"
+
+	"istio.io/bots/policybot/handlers/githubwebhook/filters"
+	"istio.io/bots/policybot/pkg/config"
+	"istio.io/bots/policybot/pkg/gh"
+	"istio.io/pkg/log"
+)
+
+// Removes the stale label from issues and prs when activity is detected
+type Unstale struct {
+	gc    *gh.ThrottledClient
+	repos map[string]bool
+}
+
+const stalenessLabel = "stale"
+const stalenessSignature = "\n\n_Courtesy of your friendly freshness tracker_."
+
+var scope = log.RegisterScope("unstale", "Removes the stale label when activity is detected on a PR or issue", 0)
+
+func NewUnstale(gc *gh.ThrottledClient, orgs []config.Org) (filters.Filter, error) {
+	u := &Unstale{
+		gc:    gc,
+		repos: make(map[string]bool),
+	}
+
+	for _, org := range orgs {
+		for _, repo := range org.Repos {
+			u.repos[org.Name+"/"+repo.Name] = true
+		}
+	}
+
+	return u, nil
+}
+
+// process an event arriving from GitHub
+func (u *Unstale) Handle(context context.Context, event interface{}) {
+	action := ""
+	orgLogin := ""
+	repoName := ""
+	number := 0
+
+	switch p := event.(type) {
+	case *github.IssuesEvent:
```

Comment:
I have to wonder, is it possible to use an interface in type switches? We could create an interface with `GetAction()`, `GetRepo()`, `GetIssue()` and cast to that to save a lot of redundant lines.

Reply:
Good idea. I tried, but it doesn't quite line up. Certain cases use  GetIssue, certain use GetPullRequest among other things...